src/cbapi/psc/models.py

```diff
@@ -0,0 +1,127 @@
+        if self.__class__.primary_key in self._dirty_attributes.keys() or self._model_unique_id is None:
+            log.debug("Creating a new {0:s} object".format(self.__class__.__name__))
+            ret = self._cb.api_json_request(self.__class__._new_object_http_method, self.urlobject,
+                                            data=self._info)
+        else:
+            updates = {}
+            for k in self._dirty_attributes.keys():
+                updates[k] = self._info[k]
+            log.debug("Updating {0:s} with unique ID {1:s}".format(self.__class__.__name__, str(self._model_unique_id)))
+            ret = self._cb.api_json_request(self.__class__._change_object_http_method,
+                                            self._build_api_request_uri(), data=updates)
+
+        return self._refresh_if_needed(ret)
+
+    def _refresh_if_needed(self, request_ret):
+        refresh_required = True
+
+        if request_ret.status_code not in range(200, 300):
+            try:
+                message = json.loads(request_ret.text)[0]
+            except:
+                message = request_ret.text
+
+            raise ServerError(request_ret.status_code, message,
+                              result="Did not update {} record.".format(self.__class__.__name__))
+        else:
+            try:
+                message = request_ret.json()
+                log.debug("Received response: %s" % message)
+                if not isinstance(message, dict):
+                    raise ServerError(request_ret.status_code, message,
+                                      result="Unknown error updating {0:s} record.".format(self.__class__.__name__))
+                else:
+                    if message.get("success", False):
+                        if isinstance(message.get(self.info_key, None), dict):
+                            self._info = message.get(self.info_key)
+                            self._full_init = True
+                            refresh_required = False
+                        else:
+                            if self._change_object_key_name in message.keys():
+                                # if all we got back was an ID, try refreshing to get the entire record.
+                                log.debug("Only received an ID back from the server, forcing a refresh")
+                                self._info[self.primary_key] = message[self._change_object_key_name]
+                                refresh_required = True
+                    else:
+                        # "success" is False
+                        raise ServerError(request_ret.status_code, message.get("message", ""),
+                                          result="Did not update {0:s} record.".format(self.__class__.__name__))
+            except:
+                pass
+
+        self._dirty_attributes = {}
+        if refresh_required:
+            self.refresh()
+        return self._model_unique_id
+
+
+class Device(PSCMutableModel):
```

Comment:
Is this just a place holder for the new device services?

Reply:
Yes. It's copied from the existing Device object, as you said I should do.